hack/jenkins/dockerized-e2e-runner.sh

```diff
@@ -17,13 +17,21 @@
 # Save environment variables in $WORKSPACE/env.list and then run the Jenkins e2e
 # test runner inside the kubekins-test Docker image.
 
+set -o errexit
+set -o nounset
+set -o pipefail
+
 env -u HOME -u PATH -u PWD -u WORKSPACE >${WORKSPACE}/env.list
+# Grab the first uncommented out line.
+readonly METADATA_SERVER_IP=$(grep '^[0-9a-fA-F\.:]\+ \+metadata\.google\.internal' /etc/hosts | head -n1 | cut -f1 -d' ' || true)
```

Comment:
We actually need both the 10.x address (which is what serves the requests) as well as the local link 169.254.169.254 address (which is what metadata.go needs during testOnGCE())

See: https://github.com/GoogleCloudPlatform/gcloud-golang/blob/master/compute/metadata/metadata.go#L208